fire/SFFireWeatherMod.F90

```diff
@@ -0,0 +1,32 @@
+module SFFireWeatherMod
+
+  use FatesConstantsMod, only : r8 => fates_r8
+
+  implicit none
+  private 
+
+  type, abstract, public :: fire_weather
+    real(r8) :: fire_weather_index   ! fire weather index
+    contains 
+    procedure(initialize_fire_weather), public, deferred :: Init
+    procedure(update_fire_weather),     public, deferred :: Update
```

Comment:
I think I like this style of using `PascalCase` for the public facing call, but using `snake_case` for the type-specific hidden procedure.

Reply:
yeah, I like having a "private" version and a "public" version?